src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -58,9 +70,199 @@ def __init__(
+                self.get_default_savepath(),
+                options=QFileDialog.Option.DontUseNativeDialog,
+            )
+
+            if not should_save:
+                return
+
+
+            self.save_current_config_to_file(save_to_path)
+            self.update_widget()
+        except Exception:
+            logging.warning(
+                "cannot save configuration to {}".format(
+                    save_to_path
+                )
+            )
+
+    def save_current_config_to_file(self, path_to_file):
+        """
+        This save the self.proto_to_configure to path_to_file
+
+        :param path_to_file: the path to file that we are saving to.
+        """
+
+        logging.info("writing to file {}".format(path_to_file))
+        path_to_directory = os.path.dirname(path_to_file)
+        os.makedirs(path_to_directory, exist_ok=True)
+
+        with open(path_to_file, "wb") as f:
+            f.write(self.proto_to_configure.SerializeToString())
+
+    def load_proto_with_file_explorer(self):
+        """
+        loading the current protobuf through file explorer
+        """
+        try:
+            path_to_file, should_open = QFileDialog.getOpenFileName(
+                self,
+                "Select Protobufs",
+                self.get_default_savepath(),
+                options=QFileDialog.Option.DontUseNativeDialog,
+            )
+            if not should_open:
+                return
+
+            self.path_to_file = path_to_file
+            self.update_proto_from_file(path_to_file)
+
+            self.update_widget()
+        except Exception:
+            logging.warning(
+                "cannot load configuration from {}. Are you sure it is a configuration proto?".format(
+                    path_to_file
+                )
+            )
+
+    def update_widget(self):
+        """
+        The following function updates the current parameters tree based on the
+        current protobuf that is being configured (i.e. self.proto_to_configure) widget.
+        """
+
+        # refreshing widgets after the parameters is called
+        logging.info("I am updating widget?")
+        self.param_group = parametertree.Parameter.create(
+            name="params",
+            type="group",
+            children=self.config_proto_to_param_dict(self.proto_to_configure),
+        )
+
+        self.param_tree.setParameters(self.param_group, showTop=False)
+        self.param_group.sigTreeStateChanged.connect(self.__handle_parameter_changed)
+        self.param_tree.setAlternatingRowColors(False)
+
+    def get_default_savepath(self):
+        """
+        returns the default save path for the protobufs
+
+        :return: the path where we are going to save the files
+        """
+
+        path_to_file = (
+            ProtoConfigurationWidget.DEFAULT_SAVE_DIRECTORY
+            + "/default_configuration.proto"
+        )
+        # appending filename
+        return path_to_file
```

Comment:
nit: This method seems kind of pointless, since it seems like it's appending two constants that are not used elsewhere together. We should just have one constant `pathlib.Path` object to the config file. We can then use the `Path` helper function to get its directory as well.

Reply:
removed.